client/src/components/Charts/ManifestStatusPieChart/ManifestStatusPieChart.tsx

```diff
@@ -1,43 +1,299 @@
-import React from 'react';
-import { Cell, Legend, Pie, PieChart, ResponsiveContainer } from 'recharts';
+import React, { ReactElement, useCallback, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
+import { Cell, Legend, Pie, PieChart, ResponsiveContainer, Sector } from 'recharts';
 
-const data = [
+interface Entry {
+  name: string;
+  value: number;
+  searchParam: string;
+}
+
+interface Color {
+  normal: string;
+  active: string;
+}
+
+const data: Array<Entry> = [
   { name: 'Pending', value: 40, searchParam: 'pending' },
   { name: 'Scheduled', value: 39, searchParam: 'scheduled' },
   { name: 'In Transit', value: 33, searchParam: 'intransit' },
   { name: 'Ready for TSDF Signature', value: 21, searchParam: 'readyforsignature' },
 ];
 
-const COLORS = ['#0088FE', '#00C49F', '#FFBB28', '#FF8042'];
+const inactiveAlpha = '1';
+const activeAlpha = '.75';
+
+const COLORS: Array<Color> = [
+  { normal: `rgba(0, 136, 254, ${inactiveAlpha})`, active: `rgba(0, 136, 254, ${activeAlpha})` },
+  { normal: `rgba(0, 196, 159, ${inactiveAlpha})`, active: `rgba(0, 196, 159, ${activeAlpha})` },
+  { normal: `rgba(255, 187, 40, ${inactiveAlpha})`, active: `rgba(255, 187, 40, ${activeAlpha})` },
+  { normal: `rgba(255, 128, 66, ${inactiveAlpha})`, active: `rgba(255, 128, 66, ${activeAlpha})` },
+];
+
+const PIE_HOVER_OFFSET_DISTANCE = 9;
+const START_LABEL_DISTANCE = 10;
+const MID_LABEL_DISTANCE = 30;
+const RADIAN = Math.PI / 180;
+
+const calculateTrig = (midAngle: number): { sin: number; cos: number } => ({
+  sin: Math.sin(-RADIAN * midAngle),
+  cos: Math.cos(-RADIAN * midAngle),
+});
+
+const calculateCoordinates = (
+  radius: number,
+  distance: number,
+  x: number,
+  y: number,
+  sin: number,
+  cos: number
+): { x: number; y: number } => {
+  return {
+    x: x + (radius + distance) * cos,
+    y: y + (radius + distance) * sin,
+  };
+};
+
+const renderCustomLabel = (props: any): ReactElement | null => {
+  const { cx, cy, midAngle, outerRadius, value, hover, activeIndex, index } = props;
+  const { sin, cos } = calculateTrig(midAngle);
+  const label = calculateCoordinates(outerRadius, -MID_LABEL_DISTANCE, cx, cy, sin, cos);
+
+  // correct for text anchoring bottom-left on (x,y) coord
+  // a positive Y value moves the element down
+  // a positive X value moves the element right
+  const ySign = sin < 0 ? -1 : 1;
+  const dDistance = 2;
+  const dx = value.toString().length * -dDistance;
+  const dy = sin > 0 ? value.toString().length * ySign * dDistance : 0;
+
+  const labelElement = (
+    <text
+      x={hover ? label.x + 8 * cos : label.x}
+      y={hover ? label.y + 8 * sin : label.y}
+      dx={dx}
+      dy={dy}
+      fill="white"
+      dominantBaseline="central"
+    >
+      {`${value}`}
+    </text>
+  );
+
+  return activeIndex !== index ? labelElement : null;
```

Comment:
@sheckathorne I made one smaller other edit. The number of manifests are shown after the initial render whether the user is hovering or not. This ensures that two labels are not applied to the same pie slice.

Reply:
This all looks great, thank you so much for helping figure out that issue. I have no further plans for the pie chart, you can go ahead and merge.